deltachat-ios/Controller/ProfileInfoViewController.swift

```diff
@@ -132,7 +132,6 @@ class ProfileInfoViewController: UITableViewController {
 
     @objc private func doneButtonPressed(_ sender: UIBarButtonItem) {
         dcContext.displayname = displayName
-        self.dismiss(animated: false, completion: nil) // not sure if this is needed
```

Comment:
why is that not needed?

i am pretty sure you are right, but i like to understand that.

is it because of onClose() calls presentTabBarController() that discards the welcomeViewController together with the ProfileInfoViewController indirectly somehow? where does this happen? i also think, we should comment a bit on this stuff, it is not obvious at least to me :)

Reply:
I tried out, on both versions if ProfileInfoController is de-initialised, and it seems, that either way, the ProfileInfoController stays in memory. So I was right, in the term, that it doesn't make a difference :(. 

I will try to fix this. Have already an idea.

Reply:
these things turn out to be complicated. ProfileInfoController stays in memory because it is referenced somehow?

Reply:
I assume we have a retain cycle somewhere. I have to look deeper into this.

Reply:
I found the memory leak, and created a PR (based on master).  I will rebase this PR after review and go from there.